tests/unit/test_charm.py

```diff
@@ -49,95 +108,137 @@ def test_update_container_correct_config(
                 }
             },
         },
-        "dsn": f"postgres://{DB_USERNAME}:{DB_PASSWORD}@{DB_ENDPOINTS}/{harness.charm._db_name}",
+        "dsn": f"postgres://{DB_USERNAME}:{DB_PASSWORD}@{DB_ENDPOINTS}/{MODEL_NAME}_{harness.charm.app.name}",
         "courier": {
             "smtp": {"connection_uri": "smtps://test:test@mailslurper:1025/?skip_ssl_verify=true"}
         },
     }
-    mocked_sql_migration.assert_called_once()
 
     assert yaml.safe_load(harness.charm._render_conf_file()) == expected_config
 
 
-def test_update_container_correct_pebble_layer(
-    harness, mocked_kubernetes_service_patcher, mocked_sql_migration
-) -> None:
+def test_on_pebble_ready_when_missing_database_relation(harness) -> None:
```

Comment:
Isn't this the same as `test_on_pebble_ready_service_not_started_when_database_not_created`?

Reply:
that one focuses on testing the service is not started